### PR TITLE
Validate records against schema

### PR DESCRIPTION
**Pull request by pselle**
Validates records against schema.json

### CODE REVIEW

File: spec/payload_spec.js [hunk 1]
```diff
@@ -15,47 +16,29 @@ describe('payload', () => {
...
     // todo: test this across all or many fields.
+    var sampleWithBadEnv = Object.assign({}, sample, {
+      environment: 2
+    })
+
     var f = () => {
-      payload.normalize({
-        environment: 2
-      })
+      payload.normalize(sampleWithBadEnv)
```
Comment (ewindisch) [inline spec/payload_spec.js]: lets test all types? n, s, {}, and [] ? (maybe the sample has these invalid entries but it's not clear from reading)

Comment (pselle) [inline spec/payload_spec.js]: Nah, the sample is a good sample, and then bad stuff is added for tests. Good suggestion.

File: schema.json [hunk 2]
```diff
@@ -1,25 +1,79 @@
 {
```
Comment (ewindisch) [inline schema.json]: @pselle you're gonna hate me for this, but we need this implemented for Python or we need to retain schema.json until we implement on the Python side... Python is reading schema.json and deepmerging it.

Comment (pselle) [inline schema.json]: hmm, not surprising, but I should've thought of it! so it shall be done

File: spec/payload_spec.js (hunk 1 above)
Comment (pselle) [inline spec/payload_spec.js]: Ok, so for strings … I am using type casting for casting things that can be numbers to numbers, as right now we receive a goodly amount of data as strings from the agents. I plan on updating the agents to send actual numbers.

Now for strings, most anything can be coerced to a string and I only drop undefined. We good with this?

File: schema.json (hunk 2 above)
Comment (pselle) [inline schema.json]: Oh also, we probably want to coerce Numbers to integers, amirite?

Comment (pselle) [inline schema.json]: The python lib, as far as I can tell, isn't used anywhere, when I went to go look.
Am I wrong?

**Pull request merged**